front/temporal/labs/utils/gong.ts

```diff
@@ -0,0 +1,195 @@
+import { getAccessTokenFromNango } from "@app/lib/labs/transcripts/utils/helpers";
+import type { LabsTranscriptsConfigurationResource } from "@app/lib/resources/labs_transcripts_resource";
+import type { Logger } from "@app/logger/logger";
+
+export async function retrieveGongTranscripts(
+  transcriptsConfiguration: LabsTranscriptsConfigurationResource,
+  localLogger: Logger
+): Promise<string[]> {
+  // Retrieve transcripts from Gong from the last 24h
+  const fromDateTime = new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString();
+
+  if (!transcriptsConfiguration.connectionId) {
+    localLogger.error(
+      {},
+      "[retrieveGongTranscripts] No connectionId found. Skipping."
+    );
+    return [];
+  }
+
+  const gongAccessToken = await getAccessTokenFromNango(
+    "gong-dev",
+    transcriptsConfiguration.connectionId
+  );
+
+  const newTranscripts = await fetch(
+    `https://api.gong.io/v2/calls?fromDateTime=${fromDateTime}`,
+    {
+      headers: {
+        Authorization: `Bearer ${gongAccessToken}`,
+        "Content-Type": "application/json",
+      },
+    }
+  );
+
+  if (!newTranscripts.ok) {
+    localLogger.error(
+      {},
+      "[retrieveNewTranscripts] Error fetching new transcripts from Gong. Skipping."
+    );
+    return [];
+  }
+
+  const newTranscriptsData = await newTranscripts.json();
+
+  if (!newTranscriptsData || newTranscriptsData.length === 0) {
+    localLogger.info(
+      {},
+      "[retrieveNewTranscripts] No new transcripts found from Gong."
+    );
+    return [];
+  }
+
+  const fileIdsToProcess = [];
+
+  for (const call of newTranscriptsData.calls) {
+    const { id: fileId } = call;
+    if (!fileId) {
+      localLogger.error(
+        {},
+        "[retrieveNewTranscripts] call does not have an id. Skipping."
+      );
+      continue;
+    }
+
+    const history = await transcriptsConfiguration.fetchHistoryForFileId(
+      fileId
+    );
+    if (history) {
+      localLogger.info(
+        { fileId },
+        "[retrieveNewTranscripts] call already processed. Skipping."
+      );
+      continue;
+    }
+
+    fileIdsToProcess.push(fileId);
+  }
+
+  return fileIdsToProcess;
+}
+
+export async function retrieveGongTranscriptContent(
+  transcriptsConfiguration: LabsTranscriptsConfigurationResource,
+  fileId: string,
+  localLogger: Logger
+): Promise<{ transcriptTitle: string; transcriptContent: string }> {
+  type GongParticipant = {
+    speakerId: string;
+    name: string;
+    [key: string]: any;
```

Comment:
Why any here? Can we have a narrower type? It's hard for me to make sense of it as a reader of this code.

Reply:
There's a ton of stuff coming back from the Gong API, felt useless to add it there because I only need the name and speakerId. Maybe I can just remove the `[key: string]: any;` ?